punx/tests/test_issue210.py

```diff
@@ -0,0 +1,177 @@
+    # 1: write the file
+    with h5py.File(hfile, "w") as root:
+        nxentry = root.create_group("entry")
+        nxentry.attrs["NX_class"] = "NXentry"
+        nxinstrument = nxentry.create_group("instrument")
+        nxinstrument.attrs["NX_class"] = "NXinstrument"
+        nxdetector = nxinstrument.create_group("detector")
+        nxdetector.attrs["NX_class"] = "NXdetector"
+
+        nxdata = nxentry.create_group("data")
+        nxdata.attrs["NX_class"] = "NXdata"
+
+        nxdata.create_dataset("data", dtype=numpy.uint32, data=[])
+
+        nxdetector["data"] = nxdata["data"]  # make the link
+
+        nxdata["data"].attrs["target"] = nxdetector["data"].name  # bad NeXus-style link
+
+    assert pathlib.Path(hfile).exists()
+
+    # 4: check the `tree` report
+    expected = [
+        "  entry:NXentry",
+        '    @NX_class = "NXentry"',
+        "    data:NXdata",
+        '      @NX_class = "NXdata"',
+        "      data --> /entry/instrument/detector/data",
+        "    instrument:NXinstrument",
+        '      @NX_class = "NXinstrument"',
+        "      detector:NXdetector",
+        '        @NX_class = "NXdetector"',
+        "        data:NX_UINT32[0] = []",
+        '          @target = "/entry/instrument/detector/data"',
+    ]
+    mc = h5tree.Hdf5TreeView(hfile)
+    assert mc is not None
+    tree_list = mc.report()
+    assert tree_list[1:] == expected
+
+    # test nxdetector["data"] for the hardlink to "/entry/instrument/detector/data"
+    with h5py.File(hfile, "r") as h5root:
+        nxdata = h5root["/entry/data"]
+        nxdetector = h5root["/entry/instrument/detector"]
+        assert nxdata["data"] == nxdetector["data"]  # hard linked
+
+        link = nxdata.get("data", getlink=True)
+        assert isinstance(link, h5py.HardLink)
+        link = nxdetector.get("data", getlink=True)
+        assert isinstance(link, h5py.HardLink)
+
+
+def test_user_file():
+    data_path = pathlib.Path(__file__).parent.parent / "data"
+    user_file = data_path / USER_FILE
+    assert user_file.exists()
+    hfile = str(user_file)
+
+    # 2: confirm it is recognized as a NeXus file
+    tree = h5tree.Hdf5TreeView(hfile)
+    assert tree.filename.endswith(hfile)
+    assert tree.isNeXus
+
+    # 4: check the `tree` report for lines containing " data"
+    expected = [
+        # # incorrect matches
+        # '    data:NXdata',
+        # "      data --> b'/entry/instrument/detector/data'",
+        # '        data --> /entry/instrument/detector/data',
+
+        # correct matches
+        "    Metadata:NXcollection",
+        "    data:NXdata",
+        '      @NX_class = "NXdata"',
+        '      @signal = "data"',
+        "      data --> /entry/instrument/detector/data",
+        '        @signal = "data"',
+        "        data:NX_UINT32[3262,3108] = __array",
+        '          @target = "/entry/instrument/detector/data"',  # FIXME
```

Comment:
data/data/data? What is happening here? Why are there so many groups called data?

Reply:
NeXus name soup. Default name for a group of NXthingy is `thingy`.  So NXdata group is `data` by default.  Default name for the plottable data field (Y axis, image, counts, ...) is also `data`.

Also, that `#FIXME` comment can be removed now.

Reply:
Succumbing to temptation, you don't imagine that's why they call it a `data` file?